### ISSUE TITLE
webcam didn't work via ThinkPad Thunderbolt dock station 

### ISSUE THREAD

**Issue by bahtioraliev** [pop-os/pop#2009]
```
NAME="Pop!_OS"
VERSION="21.04"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 21.04"
VERSION_ID="21.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
LOGO=distributor-logo-pop-os
```

**Hardware -**
Dock station - ThinkPad Thunderbolt 3 Workstation Dock Gen 2
leptop - ThinkPad t14 g2
Wire connection - Thunderbolt 3
 
Logitech streemcam didn't work via ThinkPad Thunderbolt 3 Workstation Dock Gen 2
- The webcam recognized immediately via one of the usb ports on Lenovo ThinkPad t14
- The webcam didn't recognize vi  dock station once the PC is booted 
- In some cases, the webcam could be recognized when connected to the docking station prior to the boot/restart

**Expected behavior:**
Get the same behavior when the cam connected directly via one of the usb ports in the leptop and when the cam is connected via dock station.

**Comment by bahtioraliev** [comment]
Hello @jacobgkau,
Could you please let me know what are the plans for this issue?

**Comment by leviport** [comment]
@bahtioraliev  We don't have the required hardware to help troubleshoot this issue. The thunderbolt devices I have here for testing seem to be working fine on my Oryx Pro, so it might be specific to your setup.

Please make sure your thunderbolt device is fully authorized in the Settings > Privacy > Thunderbolt section. Beyond that, I don't know how much more advice I can give.

**Comment by bahtioraliev** [comment]
@leviport This behavior is the same for elgato docking station as well.
If you need me to collect some logs, or replicate it with dubug flags, let me know. I'm ready to cooperate.
![image](https://user-images.githubusercontent.com/76793434/144888647-eab7f104-f18b-4a7f-8b2e-1e4b4e721e03.png)

**Comment by bahtioraliev** [comment]
This bahavior happen for other webcams as well.
It looks like this issue related to thunderbolt drivers.
Now running popos 20.10 it's still the same.

**Comment by bahtioraliev** [comment]
The laptop and docking station certified for ubuntu, however, none of them working as expected. 
Please let me know how we can progress on this one.

**Comment by jacobgkau** [comment]
@bahtioraliev This is not a support ticketing system, this is a development bug tracker. Please do not add additional comments unless you have further information to provide.

---

System76 customers can reach out to [support](https://support.system76.com/) for technical assistance. For non-System76 hardware, you can seek community support on [Reddit](https://reddit.com/r/pop_os/) or [Mattermost](https://chat.pop-os.org).

**Comment by jacobgkau** [comment]
> Now running popos 20.10 it's still the same.

I assume you meant 21.10 by this? 20.10 has been end-of-support for many months.

> The laptop and docking station certified for ubuntu, however, none of them working as expected.

Do they work as expected in vanilla Ubuntu?

> It looks like this issue related to thunderbolt drivers.

> Get the same behavior when the cam connected directly via one of the usb ports in the leptop

Why do you think the issue is related to the Thunderbolt drivers if the issue also occurs when using USB without the Thunderbolt dock?

> leptop - ThinkPad t14 g2

If it's an issue with a driver for the ThinkPad, it may be more prudent to contact Lenovo about the issue.

**Comment by bahtioraliev** [comment]
> > Now running popos 20.10 it's still the same.
> 
> I assume you meant 21.10 by this? 20.10 has been end-of-support for many months.
> 
> yes 21.10
>
> > The laptop and docking station certified for ubuntu, however, none of them working as expected.
> 
> Do they work as expected in vanilla Ubuntu?
> 
> > It looks like this issue related to thunderbolt drivers.
> 
> > Get the same behavior when the cam connected directly via one of the usb ports in the leptop
> 
> Why do you think the issue is related to the Thunderbolt drivers if the issue also occurs when using USB without the Thunderbolt dock?
> 
> No, the The webcam recognized Immediately in 2 of the following cases:
> 1. You plug in the webcam after reboot.
> 2. The webcam was plugged in after reboot
>
> But after hibernation, all peripherals disconnected until you will pluged them in into the laptop ports or reboot the laptop.
>
> > leptop - ThinkPad t14 g2
> 
> If it's an issue with a driver for the ThinkPad, it may be more prudent to contact Lenovo about the issue.
> Don't think this is an issue with the drivers.

**Comment by bahtioraliev** [comment]
The same behavior happen with Ethernet connection.  
When the Ethernet connection is appear/recognized/initialized only once it has been connected prior the reboot to the docking station.
* The behavior is the same with other docking stations (elgato for example)
* I think we might need to change the title to "peripherals didn't work via Thunderbolt dock station"

++++++++++++++++++++++++++++++++++
bottom line the peripherals initialized only in case they where connect to dock station and while the laptop was connected via thunderbolt cable to the docking station prior the reboot.

**Comment by bahtioraliev** [comment]
The following issue has been resolved. 
Turning off the thunderbolt Assist Mode via bios. Right after that, all described behaviors were resolved. 
![1650872218894](https://user-images.githubusercontent.com/76793434/165080844-a5fc25f2-942e-4060-8703-c1e9342bbb1d.jpg)
